src/main/java/zd/zero/waifu/motivator/plugin/tools/AssetTools.kt

```diff
@@ -0,0 +1,55 @@
+package zd.zero.waifu.motivator.plugin.tools
+
+import com.intellij.openapi.project.Project
+import zd.zero.waifu.motivator.plugin.alert.AlertConfiguration
+import zd.zero.waifu.motivator.plugin.assets.VisualMotivationAssetProvider
+import zd.zero.waifu.motivator.plugin.assets.WaifuAssetCategory
+import zd.zero.waifu.motivator.plugin.motivation.VisualMotivationFactory
+
+object AssetTools {
+
+    private const val MAXIMUM_RETRY_ATTEMPTS = 6
+
+    fun attemptToShowCategories(
+        project: Project,
+        alertConfigurationSupplier: () -> AlertConfiguration,
+        onFailure: () -> Unit,
+        vararg categories: WaifuAssetCategory
+    ) {
+        attemptToDisplayMotivation(
+            project,
+            alertConfigurationSupplier,
+            onFailure,
+            0,
+            *categories
```

Comment:
This works similar to the [ES6 spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) if you know javascripts.